x/campaign/keeper/msg_server_initialize_mainnet.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
+	"github.com/tendermint/spn/x/campaign/types"
+	profiletypes "github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) InitializeMainnet(goCtx context.Context, msg *types.MsgInitializeMainnet) (*types.MsgInitializeMainnetResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	campaign, found := k.GetCampaign(ctx, msg.CampaignID)
+	if !found {
+		return nil, sdkerrors.Wrapf(types.ErrCampaignNotFound, "%v", msg.CampaignID)
+	}
+
+	if campaign.MainnetInitialized {
+		return nil, sdkerrors.Wrapf(types.ErrMainnetInitialized, "%v", msg.CampaignID)
+	}
+
+	// Get the coordinator ID
+	coordinatorID, found := k.profileKeeper.CoordinatorIDFromAddress(ctx, msg.Coordinator)
+	if !found {
+		return nil, sdkerrors.Wrap(profiletypes.ErrCoordAddressNotFound, msg.Coordinator)
+	}
+	if campaign.CoordinatorID != coordinatorID {
+		return nil, sdkerrors.Wrap(profiletypes.ErrCoordInvalid, fmt.Sprintf(
+			"coordinator of the campaign is %v",
+			campaign.CoordinatorID,
+		))
+	}
```

Comment:
```suggestion
	coordinatorAddress, found := k.profileKeeper.GetCoordinatorAddressFromID(ctx, campaign.CoordinatorID)
	if !found {
		return nil, spnerrors.Criticalf("coordinator not found for campaign %v", msg.CampaignID)
	}
	if msg.Coordinator != coordinatorAddress {
		return nil, sdkerrors.Wrap(profiletypes.ErrCoordInvalid, fmt.Sprintf(
			"coordinator of the campaign is %v",
			coordinatorAddress,
		))
	}
```

only a suggestion ☺️. This way, we can check the campaign coordinator invariant in the same condition

Reply:
It's a bit tricky but a campaign can not have a coordinator since we have the `MsgDeleteCoordinator` therefore, this is not an invariant